qa/tasks/cephfs/test_scrub_checks.py

```diff
@@ -443,6 +444,50 @@ def test_scrub_split_dirfrags(self):
             timeout=30
         )
 
+    def test_scrub_backtrace_divergent_newer(self):
+        """
+        The consequence of the scrub will not acquire nest locks which would update
+        the iversion for the in-memory inode prior to comparing to on-disk values.
+        """
+
+        self.fs.set_max_mds(2)
+        self.fs.wait_for_daemons()
+
+        test_path2 = "testdir1/testdir2"
+        test_path3 = f"{test_path2}/testdir3"
+        test_path4 = f"{test_path3}/testdir4"
+        test_file = f"{test_path3}/file_0"
+
+        self.mount_a.run_shell(["mkdir", "-p", test_path4])
+        self.mount_a.setfattr(test_path2, "ceph.dir.pin", "0") # no surprise migrations
+
+        # Will trigger to load ancestor inodes in mds.1
+        self.mount_a.setfattr(test_path4, "ceph.dir.pin", "1")
+        self.mount_a.run_shell_payload(f"cd {test_path4} && for i in `seq 1 10`; do touch file_$i; done;")
+
+        # Dirty the ancestor inodes in mds.0 and then the mds.1 version will become older
+        self.mount_a.run_shell(["touch", f"{test_file}"])
+        self.mount_a.run_shell_payload(f"cd {test_path3} && for i in `seq 1 32`; do mkdir dir_$i; done;")
+
+        # Flush path 'testdir3/file_0' and this will persist the backtrace to disk
+        self.fs.rank_asok(["flush_path", f"/{test_file}"], rank=0)
+
+        # Export testdir3 to mds.1
+        self.fs.rank_asok(["export", "dir", f"/{test_path3}", "1"], rank=0)
```

Comment:
this should not work because `test_path2` is pinned to rank 0? You should pin `test_path3` instead. Then call `wait_for_subtrees` before continuing.

Reply:
If this test fails now despite this export failing -- and I think it should -- then that indicates this export is not necessary after all?

Reply:
Pinning `test_path3` doesn't work. And we need to export `test_path3` to `rank 1` anyway, else I couldn't reproduce it.

I have tried to pin either `test_path3` and `test_path2` to `rank 0`, both will fail the exporting:

`2024-06-20T09:01:24.731+0800 7f2a1b200640  7 mds.0.mig export_dir Cannot export to mds.1 [dir 0x1000000024f /mydir1/mydir2/mydir3/ [2,head] auth{1=1} v=35 cv=35/35 state=1074003969|complete f(v0 m2024-06-20T08:59:10.490762+0800 13=1+12) n(v0 rc2024-06-20T08:59:10.490762+0800 14=2+12) hs=13+0,ss=0+0 | child=1 replicated=1 dirty=0 authpin=0 0x5648800d0400]: dir is export pinned`

Unless there is no any pinning in the path of `testdir1/testdir2/testdir3`.

What we need to pin here is because we want to make sure before exporting all the path `testdir1/testdir2/testdir3`'s auth is `rank 0`.

I think we can just pin them to `rank 0` first and then remove the pinning before exporting.

Reply:
> Pinning `test_path3` doesn't work. And we need to export `test_path3` to `rank 1` anyway, else I couldn't reproduce it.
> 
> I have tried to pin either `test_path3` and `test_path2` to `rank 0`, both will fail the exporting:
> 
> `2024-06-20T09:01:24.731+0800 7f2a1b200640 7 mds.0.mig export_dir Cannot export to mds.1 [dir 0x1000000024f /mydir1/mydir2/mydir3/ [2,head] auth{1=1} v=35 cv=35/35 state=1074003969|complete f(v0 m2024-06-20T08:59:10.490762+0800 13=1+12) n(v0 rc2024-06-20T08:59:10.490762+0800 14=2+12) hs=13+0,ss=0+0 | child=1 replicated=1 dirty=0 authpin=0 0x5648800d0400]: dir is export pinned`
> 
> Unless there is no any pinning in the path of `testdir1/testdir2/testdir3`.

If you pin `test_path3`, it should override the pin on `test_path1`. So this is confusing. I see you're also unpinning `test_path1` but I'm not sure why that is necessary for your test.

Perhaps the issue with pinning `test_path3` is that it triggers some lock updates to rank 1?

Perhaps instead of using pinning then: turn off the balancer and use the export asok command to move things as needed.